core/trino-main/src/main/java/io/trino/sql/planner/PartitioningProviderManager.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import io.trino.connector.CatalogName;
+import io.trino.spi.connector.ConnectorNodePartitioningProvider;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class PartitioningProviderManager
```

Comment:
Just `PartitioningManager`?

Reply:
Naa - clashes with `NodePartitioningManager` you extract it from. I would just call it `PartitioningProviders` but it somewhat suggests it is utility class. Maybe still better than `PartitioningProviderManager`. WDYT?

Reply:
It's similar to other classes like that `PageSourceManager, PageSinkManager, IndexManager`. I would leave it as is for consistency